File: test-automation-core/src/main/java/com/nortal/test/core/services/hooks/AfterSuiteHook.java
package com.nortal.test.core.services.hooks;

/**
 * Hook for running stuff after test suite finishes.
 */
public interface AfterSuiteHook {

	int DEFAULT_ORDER = 10000;

	/**
	 * Returns order in which the hooks will be run. Lower order runs first.
	 * Default order: 10000
	 *
	 * @return order
	 */
	default int afterSuitOrder() {
		return DEFAULT_ORDER;
	}

	/**
	 * This method is called on all components implementing it in a random order.
	 */
	void afterSuite();
}

File: test-automation-core/src/main/kotlin/com/nortal/test/core/services/hooks/AfterSuiteHook.kt
package com.nortal.test.core.services.hooks

/**
 * Hook for running stuff after test suite finishes.
 */
interface AfterSuiteHook {
    /**
     * Returns order in which the hooks will be run. Lower order runs first.
     * Default order: 10000
     *
     * @return order
     */
    fun afterSuitOrder(): Int {
        return DEFAULT_ORDER
    }

    /**
     * This method is called on all components implementing it in a random order.
     */
    fun afterSuite()

    companion object {
        const val DEFAULT_ORDER = 10000
    }
}